packages/backend-ai/API.md

```diff
@@ -0,0 +1,87 @@
+## API Report File for "@aws-amplify/backend-ai"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { Construct } from 'constructs';
+import { FunctionResources } from '@aws-amplify/plugin-types';
+import { ResourceProvider } from '@aws-amplify/plugin-types';
+
+declare namespace constructs {
+    export {
+        ConversationHandlerFunction,
+        ConversationHandlerFunctionProps
+    }
+}
+
+declare namespace conversation {
+    export {
+        constructs,
+        runtime
+    }
+}
+export { conversation }
+
+// @public
+class ConversationHandlerFunction extends Construct implements ResourceProvider<FunctionResources> {
+    constructor(scope: Construct, id: string, props: ConversationHandlerFunctionProps);
+    // (undocumented)
+    resources: FunctionResources;
+}
+
+// @public (undocumented)
+type ConversationHandlerFunctionProps = {
+    entry?: string;
+    models: Array<{
+        modelId: string;
+        region: string;
+    }>;
+};
+
+// @public (undocumented)
+type ConversationMessage = {
+    role: 'user' | 'assistant';
+    content: Array<ConversationMessageContentBlock>;
+};
+
+// @public (undocumented)
+type ConversationMessageContentBlock = {
+    text: string;
+};
+
+// @public (undocumented)
+type ConversationTurnEvent = {
+    conversationId: string;
+    currentMessageId: string;
+    responseMutationName: string;
+    responseMutationInputTypeName: string;
+    graphqlApiEndpoint: string;
+    modelConfiguration: {
+        modelId: string;
+        systemPrompt: string;
+        region?: string;
+    };
+    request: {
+        headers: {
+            authorization: string;
+        };
+    };
+    messages: Array<ConversationMessage>;
+};
```

Comment:
I'm guessing this is defined by AppSync? How will we manage changes in this type?

Reply:
AppSync will be sending these events.

However, conversation-transformer that will deploy lambda in default scenario is going to see this type (through dependency on new construct) and can use this definition to deploy correct event producer.
In addition.  conversation-transformer and handler are going to be part of same deps graph and deploy together, so risk of them going out of sync is zero (or near zero).

@atierian please correct me if I'm wrong here.

Reply:
Correct.

> can use this definition to deploy correct event producer.

We're not currently doing this in the transformer, but this sounds like a good idea. Made a note to explore this.